src/shared/resourcefetcher/httpResourceFetcher.ts

```diff
@@ -19,36 +45,44 @@ export class HttpResourceFetcher implements ResourceFetcher {
      * @param params Additional params for the fetcher
      * @param {boolean} params.showUrl Whether or not to the URL in log statements.
      * @param {string} params.friendlyName If URL is not shown, replaces the URL with this text.
-     * @param {string} params.pipeLocation If provided, pipes output to file designated here.
-     * @param {function} params.onSuccess Function to execute on successful request.
+     * @param {function} params.onSuccess Function to execute on successful request. No effect if piping to a location.
      */
     public constructor(
         private readonly url: string,
         private readonly params: {
             showUrl: boolean
             friendlyName?: string
-            pipeLocation?: string
             onSuccess?(contents: string): void
         }
     ) {}
 
     /**
      * Returns the contents of the resource, or undefined if the resource could not be retrieved.
      */
-    public async get(): Promise<string | undefined> {
-        try {
-            this.logger.verbose(`Loading ${this.logText()}`)
+    public async get(): Promise<string | undefined>
+    public async get(pipeLocation: string): Promise<FetcherStreams>
+    public async get(pipeLocation?: string): Promise<FetcherStreams | string | undefined> {
```

Comment:
Would it be more compelling to add the Progress to the `get` function so we can get a download bar for all fetched http resources? Could also centralize things like a cancel button.

Reply:
Potentially, though the ideal would be tying the progress with a generic stream to decouple from the fetcher class. For now though I think it'd be OK to add in an optional `progress` param until we can find a better place.